.github/workflows/python-lint.yml

```diff
@@ -0,0 +1,16 @@
+name: Python Lint
+
+on:
+  push:
+    branches: ["master"]
+  pull_request:
+    branches: ["master"]
+
+jobs:
+  lint:
+    name: Run ruff check
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: astral-sh/ruff-action@v3
```

Comment:
just leaving this here: Here is how I check linting and formatting in a single action:

```
    steps:
      - uses: actions/checkout@v4
      - uses: astral-sh/ruff-action@v3
        with:
          args: "--version"
      - run: ruff check
      - run: ruff format --check --diff
```

Note that you need to set the dummy `args: --version` to make the action do a noop at setup before it runs your 2 checks